packages/components/src/internal/components/chart/CurveFitStatsGrid.tsx

```diff
@@ -0,0 +1,346 @@
+const LINEAR_EXPORT_COLS = ['slope', 'intercept', ...STATS_EXPORT_COLS];
+const POLYNOMIAL_EXPORT_COLS = ['coefficient1', 'coefficient2', 'coefficient3', ...STATS_EXPORT_COLS];
+const NONLINEAR_EXPORT_COLS = ['min', 'max', 'asymmetry', 'inflection', ...STATS_EXPORT_COLS, 'adjustedRSquared'];
+
+const STATS_EXPORT_HEADERS = ['RSS', 'TSS', 'RMSE', 'R-Squared'];
+const LINEAR_EXPORT_HEADERS = ['Slope', 'Intercept', ...STATS_EXPORT_HEADERS];
+const POLYNOMIAL_EXPORT_HEADERS = ['Coefficient 1', 'Coefficient 2', 'Coefficient 3', ...STATS_EXPORT_HEADERS];
+const NONLINEAR_EXPORT_HEADERS = [
+    'Min',
+    'Max',
+    'Asymmetry',
+    'Inflection',
+    ...STATS_EXPORT_HEADERS,
+    'Adjusted R-Squared',
+];
+
+/**
+ * Converts CurveFitRow[] to an array of arrays as expected by convertToExcel and convertToTable
+ */
+function curveFitRowsToExportFormat(hasSeries: boolean, type: string, rows: CurveFitRow[]): (number | string)[][] {
+    let headers = hasSeries ? ['Series'] : [];
+    let cols = hasSeries ? ['series'] : [];
+    if (type === 'Polynomial') {
+        headers = headers.concat(POLYNOMIAL_EXPORT_HEADERS);
+        cols = cols.concat(POLYNOMIAL_EXPORT_COLS);
+    } else if (type === 'Linear') {
+        headers = headers.concat(LINEAR_EXPORT_HEADERS);
+        cols = cols.concat(LINEAR_EXPORT_COLS);
+    } else {
+        headers = headers.concat(NONLINEAR_EXPORT_HEADERS);
+        cols = cols.concat(NONLINEAR_EXPORT_COLS);
+    }
+
+    const exportRows = rows.map((row: CurveFitRow) => {
+        return cols.map(col => row[col]);
+    });
+    exportRows.unshift(headers);
+    return exportRows;
+}
+
+type ShapeMethod = (size: number) => string;
+type ScaleMethod<T> = (value: string) => T;
+
+/**
+ * This is a subset of what our scale objects look like in our Vis API. Only defining what we're using at this moment.
+ */
+interface PlotScale<T> {
+    scale: ScaleMethod<T>;
+}
+
+/**
+ * This is the subset of attributes from the object returned by new LABKEY.vis.Plot that we need in order to render the
+ * curve fit statistics.
+ */
+interface PlotObject {
+    scales: {
+        color: PlotScale<string>;
+        shape: PlotScale<ShapeMethod>;
+    };
+}
+
+const SHAPE_SIZE = 5; // hard coded size 5 because that's what our chart legends already do
+
+interface SeriesCellProps {
+    colorScale: ScaleMethod<string>;
+    series: string;
+    shapeScale: ScaleMethod<ShapeMethod>;
+}
+
+const SeriesCell: FC<SeriesCellProps> = memo(({ colorScale, series, shapeScale }) => {
+    const color = colorScale(series);
+    const shape = shapeScale(series)(SHAPE_SIZE);
+    return (
+        <div className="curve-fit-series-cell">
+            <svg height={15} width={15}>
+                <path d={shape} fill={color} stroke={color} transform="translate(7.5, 7.5)" />
+            </svg>
+            <span className="margin-left-small">{series}</span>
+        </div>
+    );
+});
+SeriesCell.displayName = 'SeriesCell';
```

Comment:
Nice, glad this was easy to implement as I think it is a nice touch and make it quick and easy to connect table rows and the plot series